testsuite/manualmode/src/test/java/org/wildfly/core/test/standalone/mgmt/PreparedResponseTestCase.java

```diff
@@ -20,104 +20,100 @@
+        }
         controller.stop();
+        ExtensionUtils.deleteExtensionModule(BlockerExtension.MODULE_NAME);
     }
 
-    public static void deploy(ManagementClient managementClient) throws Exception {
-        ServerDeploymentHelper helper = new ServerDeploymentHelper(managementClient.getControllerClient());
-        JavaArchive war = ShrinkWrap.create(JavaArchive.class, SLOW_STOP_JAR);
-        war.addPackage(SlowStopService.class.getPackage());
-        war.addClass(TimeoutUtil.class);
-        war.addAsServiceProvider(ServiceActivator.class, SlowStopServiceActivator.class);
-        war.addAsResource(new StringAsset("Dependencies: org.jboss.dmr, org.jboss.logging, org.jboss.as.controller, org.jboss.as.server,org.wildfly.extension.request-controller, org.jboss.as.network\n"), "META-INF/MANIFEST.MF");
-        helper.deploy(SLOW_STOP_JAR, war.as(ZipExporter.class).exportAsInputStream());
-    }
-
-    public static void undeploy(ManagementClient managementClient) throws ServerDeploymentHelper.ServerDeploymentException {
-        ServerDeploymentHelper helper = new ServerDeploymentHelper(managementClient.getControllerClient());
-        helper.undeploy(SLOW_STOP_JAR);
+    private void block(ManagementClient client) throws UnsuccessfulOperationException {
+        ModelNode blockOp = Operations.createOperation("block", PathAddress.pathAddress(SUBSYSTEM, BlockerExtension.SUBSYSTEM_NAME).toModelNode());
+        blockOp.get("block-point").set("SERVICE_STOP");
+        blockOp.get("block-time").set(SHUTDOWN_WAITING_TIME);
+        client.executeForResult(blockOp);
     }
 
     @Test
     public void reloadServer() throws Exception {
-        ManagementClient mgmtClient = getManagementClient();
-        long timeout = SHUTDOWN_WAITING_TIME + System.currentTimeMillis();
-        mgmtClient.executeForResult(Operations.createOperation(RELOAD, new ModelNode().setEmptyList()));
-        while (System.currentTimeMillis() < timeout) {
-            Thread.sleep(FREQUENCY);
-            try {
-                controller.getClient().isServerInRunningState();
-            } catch (RuntimeException ex) {
-                break;
+        try (ManagementClient managementClient = getManagementClient()) {
+            block(managementClient);
+            long timeout = SHUTDOWN_WAITING_TIME + System.currentTimeMillis();
+            managementClient.executeForResult(Operations.createOperation(RELOAD, new ModelNode().setEmptyList()));
+            while (System.currentTimeMillis() < timeout) {
+                Thread.sleep(FREQUENCY);
+                try {
+                    controller.getClient().isServerInRunningState();
+                } catch (RuntimeException ex) {
+                    break;
+                }
             }
+            Assert.assertFalse(System.currentTimeMillis() < timeout);
+            Thread.sleep(2 * FREQUENCY);
+            Assert.assertTrue(managementClient.isServerInRunningState());
```

Comment:
This seems too fragile. It wouldn't be an odd thing if this took > 1 second. I think it's valid to loop for quite a long time (say 20 secs) intermittently testing for reload completion. The RuntimeException at L103 tells you the reload has progressed, so AIUI it's not relevant to this test how long it takes from there to complete.